### PR TITLE
Add py2-numpy-c-api toolfile

### PR DESCRIPTION
**Pull request by davidlt**
This was requested here: https://github.com/cms-sw/cmsdist/issues/2994

This allows writing C extensions to Numpy within CMSSW.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

### CODE REVIEW

File: py2-numpy-toolfile.spec [hunk 1]
```diff
@@ -21,6 +21,17 @@ cat << \EOF_TOOLFILE >%i/etc/scram.d/py2-numpy.xml
 </tool>
 EOF_TOOLFILE
 
+cat << \EOF_TOOLFILE >%{i}/etc/scram.d/py2-numpy-c-api.xml
+<tool name="py2-numpy-c-api" version="@TOOL_VERSION@">
+  <client>
+    <environment name="PY2_NUMPY_BASE" default="@TOOL_ROOT@"/>
```
Comment (smuzaffar) [inline py2-numpy-toolfile.spec]: @davidlt , please use PY2_NUMPY_C_API_BASE instead of PY2_NUMPY_BASE

**Comment by davidlt** [comment]
Updated.

**Comment by smuzaffar** [comment]
please test

File: py2-numpy-toolfile.spec [hunk 2]
```diff
@@ -21,6 +21,17 @@ cat << \EOF_TOOLFILE >%i/etc/scram.d/py2-numpy.xml
 </tool>
 EOF_TOOLFILE
 
+cat << \EOF_TOOLFILE >%{i}/etc/scram.d/py2-numpy-c-api.xml
+<tool name="py2-numpy-c-api" version="@TOOL_VERSION@">
+  <client>
+    <environment name="PY2_NUMPY_C_API_BASE" default="@TOOL_ROOT@"/>
+    <environment name="INCLUDE" default="$PY2_NUMPY_C_API_BASE/lib/python@PYTHONV@/site-packages/numpy-@PY2_NUMPY_REAL_VERSION@-py@PYTHONV@-linux-@OSARCH@.egg/numpy/core/include"/>
```
Comment (smuzaffar) [inline py2-numpy-toolfile.spec]: @davidlt , wouldn't it be cleaner if we create $PY2_NUMPY_ROOT/include symlink pointing to $PY2_NUMPY_ROOT/lib/python@PYTHONV@/site-packages/numpy-%realversion-py@PYTHONV@-linux-@OSARCH@.egg/numpy/core/include?

Comment (davidlt) [reply py2-numpy-toolfile.spec]: I don't see a need for it.

Comment (smuzaffar) [reply py2-numpy-toolfile.spec]: then we can avoid
```## INITENV SET PY2_NUMPY_REAL_VERSION %{realversion}``` in the numpy spec file. As numpy spec already knows its realversion, so we can create the symlink the the numpy spec.

**Comment by smuzaffar** [comment]
please test

File: py2-numpy.spec [hunk 3]
```diff
@@ -39,3 +40,8 @@ sed -ideleteme 's|#!.*/bin/python|#!/usr/bin/env python|' \
   %{i}/lib/python*/site-packages/numpy-*/numpy/distutils/conv_template.py
   
 find %{i} -name '*deleteme' -delete
+mkdir %{i}/c-api
+PYTHONV=$(echo $PYTHON_VERSION | cut -f1,2 -d.)
+OSARCH=$(uname -m)
+[ -d  %{i}/${PYTHON_LIB_SITE_PACKAGES}/numpy-%{realversion}-py${PYTHONV}-linux-$OSARCH.egg/numpy/core ] || exit 1
+ln -s   ../${PYTHON_LIB_SITE_PACKAGES}/numpy-%{realversion}-py${PYTHONV}-linux-$OSARCH.egg/numpy/core %{i}/c-api/core
```
Comment (smuzaffar) [inline py2-numpy.spec:47]: We created symlink in numpy tool itself to catch any change in directory structure which the build of numpy itself.

**Pull request merged**